core/src/main/scala/chisel3/Module.scala

```diff
@@ -858,7 +858,8 @@ package experimental {
     private[chisel3] def createSecretIO[A <: Data](data: => A)(implicit sourceInfo: SourceInfo): A = {
       val iodef = data
       requireIsChiselType(iodef, "io type")
-      require(!isFullyClosed, "Cannot create secret ports if module is fully closed")
+      if (isFullyClosed) Builder.error(s"Cannot create secret ports into fully closed ${this.name} (from ${Builder.currentModule.get.name})")
+      if (!isIOCreationAllowed) Builder.error(s"Cannot create secret ports into ${this.name} (from ${Builder.currentModule.get.name}) if IO creation is not allowed")
```

Comment:
```suggestion
      if (!isIOCreationAllowed) Builder.error(s"Cannot create bore or tap into ${this.name} (from ${Builder.currentModule.get.name}) if IO creation is not allowed")
```

or do a try-catch to give the user the right error message in boring utils?